### PR TITLE
Feature: LOAD_BOTTOM hint

### PR DESCRIPTION
**Pull request by odesenfans**
Issue Number: N/A

## Type

- [x] feature
- [ ] bugfix
- [ ] dev (no functional changes, no API changes)
- [ ] fmt (formatting, renaming)
- [ ] build
- [ ] docs
- [ ] testing

## Breaking changes?

- [ ] yes
- [x] no

### CODE REVIEW

File: src/hints/state.rs [hunk 1]
```diff
@@ -195,6 +196,42 @@ pub fn load_edge(
...
+    let preimage_vec = preimage
+        .get(&edge_bottom)
+        .ok_or(HintError::CustomHint("Edge bottom not found in preimage".to_string().into_boxed_str()))?;
+
+    let x = preimage_vec[0];
+    let y = preimage_vec[1];
+
+    let hash_ptr = get_ptr_from_var_name(vars::ids::HASH_PTR, vm, ids_data, ap_tracking)?;
+    vm.insert_value((hash_ptr + HashBuiltin::x_offset())?, x)?;
+    vm.insert_value((hash_ptr + HashBuiltin::y_offset())?, y)?;
```
Comment (notlesh) [inline src/hints/state.rs:228]: Does `hash_ptr` need to be advanced...?

Comment (maciejka) [reply src/hints/state.rs:228]: No, it is advanced in cairo.

File: src/hints/state.rs [hunk 2]
```diff
@@ -195,6 +196,42 @@ pub fn load_edge(
...
+
+pub fn load_bottom(
+    vm: &mut VirtualMachine,
+    exec_scopes: &mut ExecutionScopes,
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+    _constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    let edge = get_relocatable_from_var_name(vars::ids::EDGE, vm, ids_data, ap_tracking)?;
+    let edge_bottom = vm.get_integer((edge + NodeEdge::bottom_offset())?)?.into_owned();
```
Comment (maciejka) [inline src/hints/state.rs:216]: Is `into_owned` necessary here?

Comment (odesenfans) [reply src/hints/state.rs:216]: No, good spot!

**Review by maciejka — APPROVED** [review]

**Pull request merged**